lib/ecto/adapters/clickhouse.ex

```diff
@@ -144,8 +146,30 @@ defmodule Ecto.Adapters.ClickHouse do
   defdelegate structure_dump(default, config), to: Ecto.Adapters.ClickHouse.Structure
 
   @impl Ecto.Adapter.Structure
-  def structure_load(_default, _config) do
-    raise "not implemented"
+  def structure_load(default, config) do
+    path = config[:dump_path] || Path.join(default, "structure.sql")
+
+    Logger.info("Loading database structure from file: #{path}")
+    {:ok, conn} = Ch.Connection.connect(config)
+
+    case File.read(path) do
+      {:ok, data} ->
+        String.split(data, ";", trim: true)
+        |> Enum.filter(&String.match?(&1, ~r/[A-Za-z]+/))
```

Comment:
Why is this filter needed?

Reply:
if the file ends on a new line we would send that new line so we just remove non sql queries from the file content

Reply:
Is it the same as this then?

```elixir
String.split(data, ";")
|> Enum.map(&String.trim/1)
|> Enum.reject(&(&1 == ""))
```

Reply:
yeah, should I replace it with your code?